packages/@react-spectrum/list/src/List.tsx

```diff
@@ -0,0 +1,139 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMRef, Node} from '@react-types/shared';
+import {GridCollection, useGridState} from '@react-stately/grid';
+import {GridKeyboardDelegate, useGrid} from '@react-aria/grid';
+import {ListItem} from './ListItem';
+import {ListLayout} from '@react-stately/layout';
+import {ListState, useListState} from '@react-stately/list';
+import listStyles from './index.css';
+import React, {useMemo} from 'react';
+import {ReusableView} from '@react-stately/virtualizer';
+import {useCollator, useLocale} from '@react-aria/i18n';
+import {useProvider} from '@react-spectrum/provider';
+import {Virtualizer, VirtualizerItem} from '@react-aria/virtualizer';
+
+
+export const ListContext = React.createContext(null);
+
+export function useListLayout<T>(state: ListState<T>) {
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let layout = useMemo(() =>
+      new ListLayout<T>({
+        estimatedRowHeight: scale === 'large' ? 48 : 32,
+        estimatedHeadingHeight: scale === 'large' ? 33 : 26,
+        padding: 0, // TODO: get from DNA
+        collator
+      })
+    , [collator, scale]);
+
+  layout.collection = state.collection;
+  layout.disabledKeys = state.disabledKeys;
+  return layout;
+}
+
+function List<T extends object>(props, ref: DOMRef<HTMLDivElement>) {
+  let {
+    shouldUseVirtualFocus,
+    transitionDuration = 0
+  } = props;
+  let domRef = useDOMRef(ref);
+  let {collection} = useListState(props);
+
+  let gridCollection = new GridCollection({
+    columnCount: 1,
+    items: [...collection].map(item => ({
+      type: 'row',
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell'
+      }]
+    }))
+  });
+
+
+  let state = useGridState({
+    ...props,
+    collection: gridCollection
+  });
+  let layout = useListLayout(state); // TODO should layout return nulls?
+
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let {direction} = useLocale();
+  let keyboardDelegate = new GridKeyboardDelegate({
+    collection: state.collection,
+    disabledKeys: state.disabledKeys,
+    ref: domRef,
+    direction,
+    collator,
+    focusMode: 'cell'
+  });
+  let {gridProps} = useGrid({
+    keyboardDelegate,
+    ref: domRef
+  }, state);
+
+  let {styleProps} = useStyleProps(props);
+
+  // This overrides collection view's renderWrapper to support hierarchy of items in sections.
+  // The header is extracted from the children so it can receive ARIA labeling properties.
+  type View = ReusableView<Node<T>, unknown>;
+  let renderWrapper = (parent: View, reusableView: View) => (
+    <VirtualizerItem
+      key={reusableView.key}
+      reusableView={reusableView}
+      parent={parent} />
+  );
```

Comment:
Will List support sections in the future? If not, I don't think we need this

Reply:
I'm fine with whichever, hopefully we find out if it will support sections

Reply:
let's remove it for now